### ISSUE TITLE
[IExpression] IOperation unit tests for conversions

### ISSUE THREAD

**Issue by mavasani** [dotnet/roslyn#17590]
- [ ] IConversionExpression

These can be present in the following binding contexts:
1. Implicit conversions
2. C# `as` operator
3. VB `Trycast`, `DirectCast` operators

Look at implementations of IConversionExpression for reference: http://source.roslyn.io/#Microsoft.CodeAnalysis/Compilation/IExpression.cs,d05cbef4ee3aa239,references

**Comment by 333fred** [comment]
We'll need to skip interpolated string conversions until IOperation is updated to support interpolated strings.

**Comment by 333fred** [comment]
Still need to add implicit conversion tests for tuples, once the Tuple api is finished.

**Comment by mavasani** [comment]
@333fred is this done?

**Comment by 333fred** [comment]
No. We still need some explicit conversion tests for VB, as well as the tuple tests.
